src/2024h2/optimize-clippy.md

```diff
@@ -0,0 +1,66 @@
+# Optimizing Clippy & linting
+(a.k.a The Clippy Performance Project)
+
+| Metadata |                                                              |
+| -------- | ------------------------------------------------------------ |
+| Owner(s) | [@blyxyas][blyxyas]                                          |
+| Teams    | [Clippy team][clippy]                                                      |
+| Status   | WIP                                                          |
+
+## Summary
+
+This is the formalization and documentation of the Clippy Performance Project, a project first talked about on [Zulip, July 2023](https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy/topic/Clippy's.20performance). As the project consists of several points and is ever-changing, this document also has a dynamic structure and the team can add points. 
+
+In short, this is an effort to optimize Clippy, and Rust's linting infrastructure with a point of view of making Clippy faster (both on CI/CD pipelines, and on devs' machines)
+
+## Motivation
+
+Clippy can take up to 2.5 times the time that a normal `cargo check` takes, and it doesn't need to be! Taking so long is expensive both in development time, and in real money.
+
+On GitHub Actions, this excessive time can equal the cost of running `cargo check` on a Linux x64 32-cores machine, on a Linux x64 2-cores machine. (A 3.3-times increase).
+
+### The status quo
+
+The current status quo is that Clippy it's mostly a pre-commit / pre-merge tool. It isn't feasible to run it on-save or in the midst of working on a feature because it greatly stops the developer workflow, distracting the developer with a mindless waste of time, instead of working on the actual feature.
```

Comment:
```suggestion
The current status quo is that Clippy is mostly a pre-commit / pre-merge tool. It isn't feasible to run it on-save or in the midst of working on a feature because it greatly stops the developer workflow, distracting the developer with a mindless waste of time, instead of working on the actual feature.
```

Reply:
tbf, in a lot of my projects I often run clippy on save even with a ton of restriction lints enabled. I just keep coding even while it's checking, which I think others do as well. So I think this is more common than this implies, even if it's not the intended purpose

Reply:
I don't think we should assume any intent or purpose other than "checking your code whenever useful and convenient". Whether that's continuous, on save, pre-commit, post-commit, pre-push or in CI is left as an exercise to the reader 😎.

Reply:
We could organize a poll on the forums and subreddits to see where do people use Clippy the most.

Reply:
I've posted a poll on Reddit, Rust's user forum & Mastodon asking how do people use Clippy with a poll. Then we can sum up the votes and see what the community thinks

This what the Reddit one looks like: https://www.reddit.com/r/rust/comments/1dxu43p/feedback_poll_where_how_do_you_use_clippy/

Reply:
Seems like, most people are running Clippy in their IDE already. So this should probably  be re-worded to say, that this is a really good motivation for us to improve the speed of Clippy.